bittide/src/Bittide/ProcessingElement.hs

```diff
@@ -41,42 +43,56 @@ processingElement ::
   ( HiddenClockResetEnable dom
   , KnownNat nBusses, 2 <= nBusses, CLog 2 nBusses <= 30) =>
   PeConfig nBusses ->
-  Vec (nBusses-2) (Signal dom (WishboneS2M (Bytes 4))) ->
-  Vec (nBusses-2) (Signal dom (WishboneM2S (32 - CLog 2 nBusses) 4 (Bytes 4)))
-processingElement config bussesIn = case config of
-  PeConfig memMapConfig initI initD ->
-    go memMapConfig initI initD
+  Circuit () (Vec (nBusses-2) (Wishbone dom 'Standard (MappedBus 32 nBusses) (Bytes 4)))
+processingElement (PeConfig memMapConfig' initI' initD') =
+  go memMapConfig' initI' initD'
```

Comment:
Why is there a  `go`? Seems like the only thing you're doing is unwrapping the record?

Reply:
`PeConfig nBusses` is a GADT, not a record :). This structure enables us to hide all type variables that are not relevant outside of `processingElement` from the API of `processingElement`.

Reply:
Hmmm, let's discuss this in the office :-). Not sure what you're getting at.

Reply:
It turns out that the go function was not necessary to hide the type variables, pattern matching on the GADT is enough.